registry/storage/catalog.go

```diff
@@ -95,3 +95,23 @@ func (reg *registry) Enumerate(ctx context.Context, ingester func(string) error)
 	return nil
 
 }
+
+func pathGreaterThan(pathX, pathY string) (b bool) {
+	splitPathX := strings.SplitN(pathX, "/", 2)
+	splitPathY := strings.SplitN(pathY, "/", 2)
+
+	if splitPathX[0] == splitPathY[0] {
+		if len(splitPathX) == 1 && len(splitPathY) == 1 {
+			return false
+		} else if len(splitPathX) == 1 && len(splitPathY) != 1 {
+			return false
+		} else if len(splitPathX) != 1 && len(splitPathY) == 1 {
+			return true
+		}
+
+		return pathGreaterThan(splitPathX[1], splitPathY[1])
```

Comment:
hmm, I stand corrected.  However if you `git apply` the following, the tests fail:
```
diff --git a/registry/storage/catalog_test.go b/registry/storage/catalog_test.go
index c125cb1..38f164f 100644
--- a/registry/storage/catalog_test.go
+++ b/registry/storage/catalog_test.go
@@ -39,6 +39,8 @@ func setupFS(t *testing.T) *setupEnv {
                "bar/e",
                "foo/d/in",
                "foo-bar/b",
+               "foo-bar/baz-bat/a",
+               "foo-bar/baz-bat/b",
                "test",
        }
 ```